routes/api.js

```diff
@@ -158,6 +159,151 @@ router.patch('/user/badge/update', authenticationHelpers.isAuthOrRedirect, (req,
     }
 });
 
+
+router.post('/leaderBoard', authenticationHelpers.isAuthOrRedirect, (req, res) => {
+    if (req.body) {
+        progress.findAll({
+            include: [localUser],
+            where: {badge: req.body.badge},
```

Comment:
Why is badge being used in where clause?
Are you ranking users based on a particular badge?

Reply:
Exactly! @Raxerz, users would be ranked badge wise.

Reply:
@lunayach this is a good idea but was this implementation discussed with the team?
@Buddhiprabha @shreyans29 @paavininanda should this be badge based or for the entire set of activities/games?

Reply:
This was the idea I suggested in the proposal. This would ensure fair scoring between those who have played more stages than those who are still in earlier levels.

Reply:
Okay

Reply:
@lunayach what happens when there are no matching badges for any of the users?
How will the leaderboard be displayed? I tried this scenario and the leaderboard dialog was not displayed

Reply:
@Raxerz  This scenario is not possible because all user must be having one badge. That is, even for a new user he is having a beginner badge.
The reason for not showing up of dialogue could be the columns of badge missing. You may try to reset database and try again.